webapp-django/crashstats/base/ga.py

```diff
@@ -0,0 +1,130 @@
+"""
+Sending Google Analytics events from the webapp.
+
+See https://developers.google.com/analytics/devguides/collection/protocol/\
+v1/devguide#page
+"""
+
+import logging
+import urlparse
+import uuid
+
+from raven.transport.threaded_requests import ThreadedRequestsHTTPTransport
+
+from django.contrib.sites.requests import RequestSite
+from django.conf import settings
+
+logger = logging.getLogger('crashstats:ga')
+
+# Uncomment for local debugging. This will print all logging on stdout.
+# logger.setLevel(logging.DEBUG)
+# import sys
+# logger.addHandler(logging.StreamHandler(sys.stdout))
+
+
+def track_api_pageview(
+    request,
+    page_title=None,
+    data_source='api',
+    client_id=None,
+    **headers
+):
+    """Convenient wrapper function geared for the API calls. This way
+    the page title is automatically guessed to something sensible."""
+    page_title = page_title or 'API ({})'.format(request.path)
+    track_pageview(
+        request,
+        page_title=page_title,
+        data_source=data_source,
+        client_id=client_id,
+        **headers
+    )
+
+
+def track_pageview(
+    request,
+    page_title,
+    data_source='web',
+    client_id=None,
+    **headers
+):
+    """Trigger a HTTP POST to Google Analytics about a particular page being
+    viewed. Most paramters are being picked out of the request object.
+
+    :arg request: The Django request object.
+    :arg page_title: A string to act as a title for the page. Not relevant
+    or particularly needed for API requests.
+    :arg data_source: A string like 'web' or 'api'. See documentation.
+    :arg client_id: Some string to identify the client.
+
+    """
+    if not settings.GOOGLE_ANALYTICS_ID:
+        logging.debug('GOOGLE_ANALYTICS_ID not set up. No pageview tracking.')
+        return
+
+    domain = settings.GOOGLE_ANALYTICS_DOMAIN
+    if not domain or domain == 'auto':
+        domain = RequestSite(request).domain
+
+    params = {}
+    params['v'] = 1  # version
+    params['tid'] = settings.GOOGLE_ANALYTICS_ID  # Tracking ID / Property ID
+    params['dh'] = domain
```

Comment:
We might as well send dl since we're sending all the parts.

https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#dl